indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/Asn1Elements.kt

```diff
@@ -235,26 +255,25 @@ sealed class Asn1Element(
 
     @Serializable
     @ConsistentCopyVisibility
-    data class Tag private constructor(
-        val tagValue: ULong, val encodedTagLength: Int,
+    data class Tag internal constructor(
+        val tagValue: ULong,
         @Serializable(with = ByteArrayBase64Serializer::class) val encodedTag: ByteArray
     ) : Comparable<Tag> {
-        private constructor(values: Triple<ULong, Int, ByteArray>) : this(values.first, values.second, values.third)
-        constructor(derEncoded: ByteArray) : this(
-            derEncoded.iterator().decodeTag().let { Triple(it.first, it.second.size, derEncoded) }
-        )
 
+
+        constructor(derEncoded: ByteArray) : this(derEncoded.iterator().decodeTag())
+
+        //workaround because we cannot return two values or assign params in a destructured manner
```

Comment:
unclear what this means

Reply:
we need to "directly" pass the result of `decodeTag` to a constructor. which is a pair. you cannot destructure the return value directly in a function or constructor invocation, so we need another constructor to do the destructuring.

Reply:
i would prefer something like `decodeTag(...).let { (a,b) -> Tag(a,b) }`, though I am also unsure why `decodeTag` doesn't return a `Tag` in the first place